coalib/bearlib/__init__.py

```diff
@@ -82,34 +97,52 @@ def _deprecate_decorator(func):
         logged_deprecated_args = set()
 
         def wrapping_function(*args, **kwargs):
-            for arg, depr_arg_and_modifier in depr_args.items():
-                deprecated_arg, _func = (
-                    depr_arg_and_modifier
-                    if isinstance(depr_arg_and_modifier, tuple)
-                    else (depr_arg_and_modifier, lambda x: x))
+            for arg, depr_value in wrapping_function.__metadata__.depr_values:
+                if isinstance(depr_value, dict):
+                    modifier = list(depr_value.values())[0]
+                    deprecated_arg = list(depr_value.keys())[0]
+                else:
+                    deprecated_arg = depr_value
+                    modifier = (
+                        wrapping_function.__metadata__.default_modifier)
                 if deprecated_arg in kwargs:
                     if deprecated_arg not in logged_deprecated_args:
                         logging.warning(
-                            'The setting `{}` is deprecated. Please use `{}` '
-                            'instead.'.format(deprecated_arg, arg))
+                            'The setting `{}` is deprecated. Please use'
+                            ' `{}` instead.'.format(deprecated_arg, arg))
                         logged_deprecated_args.add(deprecated_arg)
-                    depr_arg_value = _func.__call__(kwargs[deprecated_arg])
+                    depr_arg_value = modifier.__call__(
+                        kwargs[deprecated_arg])
                     if arg in kwargs and depr_arg_value != kwargs[arg]:
                         logging.warning(
                             'The value of `{}` and `{}` are conflicting.'
                             ' `{}` will be used instead.'.format(
-                                  deprecated_arg, arg, arg))
+                                deprecated_arg, arg, arg))
                     else:
                         kwargs[arg] = depr_arg_value
                     del kwargs[deprecated_arg]
+
```

Comment:
while I do prefer a blank line here, I'd much prefer that this patch is shorter so it is easier to see what is being changed.